components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -137,6 +153,8 @@ struct esp_websocket_client {
     int                         payload_offset;
     esp_transport_keep_alive_t  keep_alive_cfg;
     struct ifreq                *if_name;
+    int                         memory_type;
```

Comment:
I'm not against making the selection in runtime but, in other components, we make this choice in compile time through Kconfig. 
Do you have a use case to have it set in runtime?